### DISCUSSION TITLE
Need help in deploying live helper chat

### DISCUSSION THREAD

**Discussion by Xiddiqui** [LiveHelperChat/livehelperchat#2090] [Q&A]
I have done some changes  in design/defaulttheme/widget/react-app/  and i installed some packages too. its working fine on locally but when i want to deploy it to to server i am unable to reflect changes . I am stuck need help

**Comment by Shamsafayyaz** [comment]
i'm facing the same issue

**Comment by remdex** [comment]
You are not following quick development guide
https://doc.livehelperchat.com/docs/development/quick-guide

After you recompiled the app and it's working fine on your env. URL Generated by LHC is heavily cached by browser CDN and so on. Because URL remains the same.
```
https://demo.livehelperchat.com/design/defaulttheme/js/widgetv2/react.app.js?v=325
```

After your changes you can increase those two numbers. So on production after clearing cache new URL will be generated for React application. If your app is not in production it does not matter because all browserse will download new version of a file in any case. But if you are doing updates and want them to be presented as soon you have to do this. E.g have your own fork.
```
https://github.com/LiveHelperChat/livehelperchat/blob/master/lhc_web/modules/lhwidgetrestapi/settings.php#L538-L541
```

Here is sample commit where I do those things - https://github.com/LiveHelperChat/livehelperchat/commit/223536bfd2cbc603c79f339f6ec8c950d3a79596

If you do not want to modify original files you can listen for custom event in your extension and modify there. See any extension how they listen for events.
```
erLhcoreClassChatEventDispatcher::getInstance()->dispatch('widgetrestapi.settings', array('output' => & $outputResponse));
```



Also it's advised after you recompile just in case run. If you have some extension activated run this command. It will generate static CSS and JS files and then you can use multiple http instances as JS and CSS files requested always will be the same.
```
php cron.php -s site_admin -c cron/util/generate_css -p 1 && gulp js-static
```

Pull request for improved documentation is always welcome :)
